File: python/Lab10/binary.py
class Binary():
    def __init__(self, number = None) -> None:
        if number != None:
            try:
                for i in number:
                    if i != "1" and i != "0":
                        raise ValueError("Input string is not a sequence of digits.")
                self.binary_number = int(number, 2)
                self.convert_status = "Successfully converted"
            except ValueError:
                print("Input string is not a sequence of digits.")
                self.convert_status = "Error! String isn't a binary digit!"
            except OverflowError:
                print("The number cannot fit in an bin.")
                self.convert_status = "Error! Cannot fit in format!"
        else: self.convert_status = None
        self.binary_status = None
        
    
    @property
    def get_binary_status(self):
        return self.binary_status 
    @property
    def get_convert_status(self):
        return self.convert_status 

    def input_binary(self):
        while(True):
            try:
                number = input("Input binary number : ")
                for i in number:
                    if i != "1" and i != "0":
                        raise ValueError("Input string is not a sequence of digits.")
                self.binary_number = int(number, 2)
                self.convert_status = "Successfully converted"
            except ValueError:
                print("Input string is not a sequence of digits.")
                self.convert_status = "Error! String isn't digit!"
            except OverflowError:
                print("The number cannot fit in an bin.")
                self.convert_status = "Error! Cannot fit in format!"
            else: break
    
    def logic_multiplication(self, number):
        result = -48753975
        if self.convert_status == "Successfully converted":
            result = self.binary_number & number.binary_number
            self.binary_status = "The number is logically multiplied"
            return result
        else:
            print("Error! The number is not in correct format!")
    
    def logic_addition(self, number):
        result = -48753975
        if self.convert_status == "Successfully converted":
            result = self.binary_number | number.binary_number
            self.binary_status = "The number is logically added"
            return result
        else:
            print("Error! The number is not in correct format!")
    
    def XOR(self, key):
        result = -48753975
        if self.convert_status == "Successfully converted":
            result = self.binary_number ^ key.binary_number
            self.binary_status = "The number is encrypted"
            return result
        else:
            print("Error! The number is not in correct format!")
    
    def inverse(self):
        result = -48753975
        if self.convert_status == "Successfully converted":
            result = ~self.binary_number
            self.binary_status = "The number is inversed"
            return result
        else:
            print("Error! The number is not in correct format!")
    
    def twos_complement(self):
        result = -48753975
        if self.convert_status == "Successfully converted":
            result = ~self.binary_number + 1
            self.binary_status = "Number converted to two's complement"
            return result
        else:
            print("Error! The number is not in correct format!")

    def left_shift(self, shift):
        result = -48753975
        if self.convert_status == "Successfully converted":
            result = self.binary_number << shift
            self.binary_status =  f"The number shifted left {shift} places"
            return result
        else:
            print("Error! The number is not in correct format!")

    def right_shift(self, shift):
        result = -48753975
        if self.convert_status == "Successfully converted":
            result = self.binary_number >> shift
            self.binary_status =  f"The number shifted right {shift} places"
            return result
        else:
            print("Error! The number is not in correct format!")
    
    def addition(self, number):
        result = -48753975
        if self.convert_status == "Successfully converted":
            result = self.binary_number + number.binary_number
            self.binary_status = f"The number is added to {number.binary_number}"
            return result
        else:
            print("Error! The number is not in correct format!")  

    def subtraction(self, number):
        result = -48753975
        if self.convert_status == "Successfully converted":
            result = self.binary_number - number.binary_number
            self.binary_status = f"{number.binary_number} units are substracted from the number"
            return result
        else:
            print("Error! The number is not in correct format!")
    
    def multiplication(self, number):
        result = -48753975
        if self.convert_status == "Successfully converted":
            result = self.binary_number * number.binary_number
            self.binary_status = f"The number is multiplied by {number.binary_number}"
            return result
        else:
            print("Error! The number is not in correct format!")
    
    def division(self, number):
        result = -48753975
        if self.convert_status == "Successfully converted":
            result = self.binary_number / number.binary_number
            self.binary_status = f"The number is divided by {number.binary_number}"
            return result
        else:
            print("Error! The number is not in correct format!")
    

File: c#/Lab10/Lab10_2/Binary.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Lab10_2
{
    class Binary
    {
        private int binaryNumber;
        public int BinaryNumber
        {
            set
            {
                try
                {
                    this.binaryNumber = Convert.ToInt32(value.ToString(), 2);
                    this.convertStatus = "Successfully converted";
                }
                catch (FormatException)
                {
                    Console.WriteLine("Input string is not a sequence of digits.");
                    this.convertStatus = "Error! String isn't digit!";
                }
                catch (OverflowException)
                {
                    Console.WriteLine("The number cannot fit in an Int32.");
                    this.convertStatus = "Error! Cannot fit in format!";
                }
            }
            get
            {
                return this.binaryNumber;
            }
        }

        private string convertStatus;
        public string ConvertStatus { get { return this.convertStatus; } }

        private string binaryStatus;
        public string BinaryStatus { get { return this.binaryStatus; } }

        public Binary()
        {

        }
        public Binary(string binary)
        {
            try
            {
                this.binaryNumber = Convert.ToInt32(binary, 2);
                this.convertStatus = "Successfully converted";
            }
            catch (FormatException)
            {
                Console.WriteLine("Input string is not a sequence of digits.");
                this.convertStatus = "Error! String isn't digit!";
            }
            catch (OverflowException)
            {
                Console.WriteLine("The number cannot fit in an Int32.");
                this.convertStatus = "Error! Cannot fit in format!";
            }
        }
        
        public void InputBinary()
        {
            while (true)
            {
                Console.Write("Input binary number : ");
                var i = 0;
                try
                {
                    this.binaryNumber = Convert.ToInt32(Console.ReadLine(), 2);
                    this.convertStatus = "Successfully converted";
                    i++;
                }
                catch (FormatException)
                {
                    Console.WriteLine("Input string is not a sequence of digits.");
                    this.convertStatus = "Error! String isn't digit!";
                }
                catch (OverflowException)
                {
                    Console.WriteLine("The number cannot fit in an Int32.");
                    this.convertStatus = "Error! Cannot fit in format!";
                }
                if (i != 0) break;
            }
        }

        public int LogicMultiplication(Binary number)
        {
            int result = -48753975;
            if (this.convertStatus == "Successfully converted")
            {
                result = this.binaryNumber & number.binaryNumber;
                this.binaryStatus = "The number is logically multiplied";
            }
            else
            {
                Console.WriteLine("Error! The number is not in correct format!");
            }
            return result;
        }

        public int LogicAddition(Binary number)
        {
            int result = -48753975;
            if (this.convertStatus == "Successfully converted")
            {
                result = this.binaryNumber | number.binaryNumber;
                this.binaryStatus = "The number is logically added";
            }
            else
            {
                Console.WriteLine("Error! The number is not in correct format!");
            }
            return result;
        }

        public int XOR(Binary key)
        {
            int result = -48753975;
            if (this.convertStatus == "Successfully converted")
            {
                result = this.binaryNumber ^ key.binaryNumber;
                this.binaryStatus = "The number is encrypted";
            }
            else
            {
                Console.WriteLine("Error! The number is not in correct format!");
            }
            return result;
        }
        
        public int Inverse()
        {
            int result = -48753975;
            if (this.convertStatus == "Successfully converted")
            {
                result = ~this.binaryNumber;
                this.binaryStatus = "The number is inversed";
            }
            else
            {
                Console.WriteLine("Error! The number is not in correct format!");
            }
            return result;
        }

        public int TwosComplement()
        {
            int result = -48753975;
            if (this.convertStatus == "Successfully converted")
            {
                result = ~this.binaryNumber + 1;
                this.binaryStatus = "Number converted to two's complement";
            }
            else
            {
                Console.WriteLine("Error! The number is not in correct format!");
            }
            return result;
        }
        
        public int LeftShift(int count)
        {
            int result = -48753975;
            if (this.convertStatus == "Successfully converted")
            {
                result = this.binaryNumber << count;
                this.binaryStatus = $"The number shifted left {count} places";
            }
            else
            {
                Console.WriteLine("Error! The number is not in correct format!");
            }
            return result;
        }
       
        public int RightShift(int count)
        {
            int result = -48753975;
            if (this.convertStatus == "Successfully converted")
            {
                result = this.binaryNumber >> count;
                this.binaryStatus = $"The number shifted right {count} places";
            }
            else
            {
                Console.WriteLine("Error! The number is not in correct format!");
            }
            return result;
        }
        
        public int Addition(Binary number)
        {
            int result = -48753975;
            if (this.convertStatus == "Successfully converted")
            {
                result = this.binaryNumber + number.binaryNumber;
                this.binaryStatus = $"The number is added to {number.binaryNumber}";
            }
            else
            {
                Console.WriteLine("Error! The number is not in correct format!");
            }
            return result;
        }
        
        public int Subtraction(Binary number)
        {
            int result = -48753975;
            if (this.convertStatus == "Successfully converted")
            {
                result = this.binaryNumber - number.binaryNumber;
                this.binaryStatus = $"{number.binaryNumber} units are substracted from the number";
            }
            else
            {
                Console.WriteLine("Error! The number is not in correct format!");
            }
            return result;
        }
        
        public int Multiplication(Binary number)
        {
            int result = -48753975;
            if (this.convertStatus == "Successfully converted")
            {
                result = this.binaryNumber * number.binaryNumber;
                this.binaryStatus = $"The number is multiplied by {number.binaryNumber}";
            }
            else
            {
                Console.WriteLine("Error! The number is not in correct format!");
            }
            return result;
        }
        
        public int Division(Binary number)
        {
            int result = -48753975;
            if (this.convertStatus == "Successfully converted")
            {
                result = this.binaryNumber / number.binaryNumber;
                this.binaryStatus = $"The number is divided by {number.binaryNumber}";
            }
            else
            {
                Console.WriteLine("Error! The number is not in correct format!");
            }
            return result;
        }
    }
}
